types/src/schemas/stoatConfigSchemaRendered.json

```diff
@@ -121,6 +128,51 @@
         }
       }
     },
+    "static_hosting_plugin_rendered_variants": {
+      "type": "object",
+      "required": ["with_variants", "path", "sha", "variants"],
+      "description": "Static hosting tasks with variants.",
+      "properties": {
+        "metadata": {
+          "type": "object",
+          "additionalProperties": true
+        },
+        "with_variants": {
+          "type": "boolean",
+          "const": true
+        },
+        "path": {
+          "type": "string"
+        },
+        "file_viewer": {
+          "type": "boolean"
+        },
+        "sha": {
+          "type": "string"
+        },
+        "variants": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "object",
+            "description": "The field name is the normalized variant name.",
```

Comment:
Why is normalization required? Could we just have the actual variant name here and then omit it from the value object? The actual variant name is also ordered by sorting and has both the key and value which makes it safer from re-ordering and re-naming ending up with conflicts in some edge cases. 

Is it because we want to have flexibility to change the naming format without breaking partial config mergeability? If that's the case, we should probably use the same format as the full name now as the key and have it duplicated across the key and the `.variant` field in the value.

Reply:
> Why is normalization required? Could we just have the actual variant name here and then omit it from the value object? The actual variant name is also ordered by sorting and has both the key and value which makes it safer from re-ordering and re-naming ending up with conflicts in some edge cases.

Just an old habit to avoid any special character in the field name, which is unnecessary. Will change it to variant key value pairs separated by comma.

> Is it because we want to have flexibility to change the naming format without breaking partial config mergeability? If that's the case, we should probably use the same format as the full name now as the key and have it duplicated across the key and the .variant field in the value.

It should not change the mergability. I think we can remove the `variant` field for now.